state-chain/pallets/cf-trading-strategy/src/lib.rs

```
@@ -250,6 +251,14 @@ pub mod pallet {
 		) -> DispatchResult {
 			let lp = &T::AccountRoleRegistry::ensure_liquidity_provider(origin)?;
 
+			// Check that the pools exist
+			for (base, quote) in strategy.supported_asset_pairs() {
+				if !T::PoolApi::pool_exists(base, quote) {
+					return Err(Error::<T>::PoolDoesNotExist.into());
```

Do we have to verify this here?

Isn't it fine if we create a strategy that then just can't add orders because the pool doesn't exist? 

What if the pool exists when the strategy is created, but is deleted after? (Unlikely I agree, but so is creating a strategy for non-existent pools).

---

I put it here because I was hitting the `log_or_panic` in the idle function when trying to create the limit orders.
But you are right, the pool could be deleted, so we should change it from a `log_or_panic` to an expected error case.
I still think that we should check that the pool exists here and give early feedback to the user.

---

I feel like we should return an error when it's an error, not just as a convenience. I see little added value in checking this her since it's (a) very unlikely to happen and (b) has no real consequence except the strategy might not do anything and needs to be canceled. 

I don't think the benefit justifies the added trait method/error case.